crates/cast/src/lib.rs

```diff
@@ -159,23 +161,27 @@ where
     ///
     /// # Example
     ///
-    /// ```ignore
-    /// use cast::{Cast, TxBuilder};
-    /// use ethers_core::types::Address;
-    /// use ethers_providers::{Http, Provider};
+    /// ```
+    /// use cast::{Cast};
+    /// use alloy_primitives::{Address, U256, Bytes};
+    /// use alloy_rpc_types::{TransactionRequest, WithOtherFields};
+    /// use foundry_common::provider::alloy::get_http_provider;
     /// use std::str::FromStr;
+    /// use alloy_sol_types::{sol, SolCall};
+    ///
+    /// sol!(
+    /// function greeting(uint256 i) public returns (string);
+    /// );
     ///
     /// # async fn foo() -> eyre::Result<()> {
-    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
+    /// let provider = get_http_provider("http://localhost:8545");
```

Comment:
this should not be part of the docs, this is an internal helper function

the example should just use a regular alloy provider